test/e2e/rds/tests/test_db_subnet_group.py

```diff
@@ -0,0 +1,109 @@
+from rds.replacement_values import REPLACEMENT_VALUES
+from common.resources import load_resource_file, random_suffix_name
+from common import k8s
+
+RESOURCE_PLURAL = 'dbsubnetgroups'
+
+DELETE_WAIT_AFTER_SECONDS = 20
+CREATE_INTERVAL_SLEEP_SECONDS = 15
+# Time to wait before we get to an expected RUNNING state.
+# In my experience, it regularly takes more than 6 minutes to create a
+# single-instance RabbitMQ broker...
+CREATE_TIMEOUT_SECONDS = 600
+
+
+@pytest.fixture(scope="module")
+def rds_client():
+    return boto3.client('rds')
+
+
+@service_marker
+@pytest.mark.canary
+class TestDBSubnetgroup:
+    def test_create_delete_2az(self, rds_client):
+        resource_name = "my-subnet-group"
+        resource_desc = "my-subnet-group description"
+
+        br_resources = get_bootstrap_resources()
+
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["DB_SUBNET_GROUP_NAME"] = resource_name
+        replacements["DB_SUBNET_GROUP_DESC"] = resource_desc
+        replacements["SUBNET_AZ1"] = br_resources.SubnetAZ1
+        replacements["SUBNET_AZ2"] = br_resources.SubnetAZ2
+
+        resource_data = load_resource_file(
+            SERVICE_NAME,
+            "db_subnet_group_2az",
+            additional_replacements=replacements,
+        )
+        logging.debug(resource_data)
+
+        # Create the k8s resource
+        ref = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+            resource_name, namespace="default",
+        )
+        k8s.create_custom_resource(ref, resource_data)
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        assert cr is not None
+        assert k8s.get_resource_exists(ref)
+
+        # Let's check that the DB subnet group appears in RDS
+        aws_res = rds_client.describe_db_subnet_groups(DBSubnetGroupName=resource_name)
+        assert aws_res is not None
+        assert len(aws_res['DBSubnetGroups']) == 1
+
+        now = datetime.datetime.now()
+        timeout = now + datetime.timedelta(seconds=CREATE_TIMEOUT_SECONDS)
+
+        # TODO(jaypipes): Move this into generic AWS-side waiter
+        while aws_res['DBSubnetGroups'][0]['SubnetGroupStatus'] != "Complete":
+            if datetime.datetime.now() >= timeout:
+                raise Exception("failed to find DB subnet group in Complete status before timeout")
+            time.sleep(CREATE_INTERVAL_SLEEP_SECONDS)
+            aws_res = rds_client.describe_db_subnet_groups(DBSubnetGroupName=resource_name)
+            assert aws_res is not None
+            assert len(aws_res['DBSubnetGroups']) == 1
+
+        # Delete the k8s resource on teardown of the module
+        k8s.delete_custom_resource(ref)
+
+        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+        # DB subnet group should no longer appear in RDS
+        res_found = False
+        try:
+            aws_res = rds_client.describe_db_subnet_groups(DBSubnetGroupName=resource_name)
+            res_found = True
```

Comment:
why not just `assert False` in this line, rather than messing with a boolean?

Reply:
copy/paste from my AmazonMQ tests :) I'll fix, nice callout.